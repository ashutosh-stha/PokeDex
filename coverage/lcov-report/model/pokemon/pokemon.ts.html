
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for model/pokemon/pokemon.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">model/pokemon</a> pokemon.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>7/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">66.66% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/3</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>7/10</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {createModel} from '@rematch/core';
import {RootModel} from '../models';
&nbsp;
export interface Pokemon {
  name: string;
  url: string;
}
&nbsp;
interface PokemonResponse {
  results: Pokemon[];
}
&nbsp;
interface PokemonState {
  pokemonList: Pokemon[];
}
&nbsp;
const initialState: PokemonState = {
  pokemonList: [],
};
&nbsp;
export const pokemon = createModel&lt;RootModel&gt;()({
  state: initialState,
  reducers: {
<span class="fstat-no" title="function not covered" >    se</span>tPokemonList(state, payload) {
<span class="cstat-no" title="statement not covered" >      return {...state, pokemonList: payload};</span>
    },
  },
  effects: dispatch =&gt; ({
    async loadPokemon(_, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonData = await apiService?.get&lt;PokemonResponse&gt;(
          '/pokemon?offset=30&amp;limit=30',
        );
        <span class="missing-if-branch" title="if path not taken" >I</span>if (pokemonData) {
<span class="cstat-no" title="statement not covered" >          dispatch.pokemon.setPokemonList(pokemonData.results);</span>
        }
      } catch (error) {
        // Handle any errors here
<span class="cstat-no" title="statement not covered" >        console.error('Failed to load Pokemon:', error);</span>
      }
    },
  }),
});
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-06-18T10:44:52.181Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    