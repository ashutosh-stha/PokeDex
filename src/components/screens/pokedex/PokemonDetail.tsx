import React, {FC, useEffect} from 'react';
import {Image, SafeAreaView, StyleSheet, Text, View} from 'react-native';
import {Route, useRoute} from '@react-navigation/native';
import {useDispatch, useSelector} from 'react-redux';
import {Dispatch, RootState} from '../../../model/store';
import {LoadingComponent} from '../../../commonComponents/loadingComponent/LoadingComponent';

interface Props {
  pokemon?: string;
}

export const PokemonDetail: FC = () => {
  const dispatch = useDispatch<Dispatch>();
  const pokemonDetail = useSelector(
    (state: RootState) => state.pokemon?.pokemonDetail,
  );
  const pokemonDetailLoading = useSelector(
    (state: RootState) => state.loading.effects.pokemon.loadPokemonDetail,
  );
  const {params} = useRoute<Route<'Pokemon', Props>>();

  useEffect(() => {
    dispatch.pokemon.loadPokemonDetail(params.pokemon);
    return () => {
      dispatch.pokemon.setPokemonDetail(undefined);
    };
  }, [dispatch.pokemon, params.pokemon]);

  return (
    <SafeAreaView style={styles.container}>
      {pokemonDetailLoading ? (
        <LoadingComponent />
      ) : (
        <View style={styles.container}>
          <View style={styles.horizontalContainer}>
            <Image
              style={styles.defaultImage}
              source={{
                uri: `${pokemonDetail?.sprites.front_default}`,
              }}
            />
            <View
              style={[styles.verticalContainer, styles.descriptionContainer]}>
              <Text style={styles.title}>{pokemonDetail?.name}</Text>
              <Text style={styles.subHeader}>Type</Text>
              <View style={styles.horizontalContainer}>
                {pokemonDetail?.types.map(({type}) => {
                  return (
                    <Text style={styles.types} key={type.name}>
                      {type.name}
                    </Text>
                  );
                })}
              </View>
            </View>
          </View>
        </View>
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  defaultImage: {
    width: 150,
    height: 150,
    resizeMode: 'stretch',
  },
  horizontalContainer: {
    flexDirection: 'row',
  },
  verticalContainer: {
    flexDirection: 'column',
  },
  descriptionContainer: {
    justifyContent: 'center',
    margin: 16,
  },
  title: {
    textTransform: 'capitalize',
    fontWeight: 'bold',
    fontSize: 20,
    marginBottom: 10,
  },
  subHeader: {
    fontSize: 14,
    fontWeight: 'bold',
  },
  types: {
    marginTop: 6,
    marginRight: 8,
    textTransform: 'capitalize',
  },
});
