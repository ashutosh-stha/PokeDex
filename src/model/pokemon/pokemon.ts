import {createModel} from '@rematch/core';
import {RootModel} from '../models';

export interface Pokemon {
  name: string;
  url: string;
}

interface PokemonResponse {
  results: Pokemon[];
}

export type PokemonType = {
  name: string;
};
type Sprites = {
  back_default: string | null;
  back_female: string | null;
  back_shiny: string | null;
  back_shiny_female: string | null;
  front_default: string | null;
  front_female: string | null;
  front_shiny: string | null;
  front_shiny_female: string | null;
  other: {
    dream_world: {
      front_default: string | null;
      front_female: string | null;
    } | null;
    home: {
      front_default: string | null;
      front_female: string | null;
      front_shiny: string | null;
      front_shiny_female: string | null;
    } | null;
    'official-artwork': {
      front_default: string | null;
      front_shiny: string | null;
    } | null;
  };
};

interface PokemonDetail {
  id: number;
  name: string;
  types: {type: PokemonType}[];
  sprites: Sprites;
}

interface PokemonState {
  pokemonList: Pokemon[];
  pokemonDetail?: PokemonDetail;
}

const initialState: PokemonState = {
  pokemonList: [],
  pokemonDetail: undefined,
};
export const pokemon = createModel<RootModel>()({
  state: initialState,
  reducers: {
    setPokemonList(state, payload) {
      return {...state, pokemonList: payload};
    },
    setPokemonDetail(state, payload) {
      return {...state, pokemonDetail: payload};
    },
  },
  effects: dispatch => ({
    async loadPokemon(_, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonData = await apiService?.get<PokemonResponse>(
          '/pokemon?offset=100&limit=100',
        );
        if (pokemonData) {
          dispatch.pokemon.setPokemonList(pokemonData.results);
        }
      } catch (error) {
        // Handle any errors here
        console.error('Failed to load Pokemon:', error);
      }
    },
    async loadPokemonDetail(pokemonName, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonDetail = await apiService?.get<PokemonDetail>(
          `/pokemon/${pokemonName}`,
        );
        if (pokemonDetail) {
          dispatch.pokemon.setPokemonDetail(pokemonDetail);
        }
      } catch (error) {
        console.error('Failed to load Pokemon Detail:', error);
      }
    },
  }),
});
