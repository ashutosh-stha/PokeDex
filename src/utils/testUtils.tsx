import React from 'react';
import {Provider} from 'react-redux';
import {Store} from 'redux';
import {init} from '@rematch/core';
import {render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';

type WrapWithStore = {
  children: any;
};
type InitPropsType = {
  wrapper?: any;
};
const wrapWithStore = (store: Store) => {
  return ({children}: WrapWithStore) => (
    <Provider store={store}>
      <NavigationContainer>{children}</NavigationContainer>
    </Provider>
  );
};
class ComponentRenderer {
  private component;
  private models: any;
  private store: any;
  constructor(component: any) {
    this.component = component;
  }

  withModels(models: any) {
    this.models = models;
    return this;
  }
  render() {
    const initProps: InitPropsType = {};
    if (this.models) {
      initProps.wrapper = wrapWithStore(init({models: this.models}));
    }
    return render(this.component, initProps);
  }
}
export function componentRenderer(component: any) {
  return new ComponentRenderer(component);
}
