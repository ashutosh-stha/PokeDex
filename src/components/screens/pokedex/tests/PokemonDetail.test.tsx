import React from 'react';
import {componentRenderer} from '../../../../utils/testUtils';
import {PokemonDetail as PokemonDetailType} from '../../../../model/pokemon/pokemon';
import {PokemonDetail} from '../PokemonDetail';

const POKEMON_DETAIL: PokemonDetailType = {
  name: 'Pikachu',
  id: 1,
  types: [{type: {name: 'electric'}}],
  sprites: {front_default: 'url'},
};

const MOCK_LOAD_POKEMON_DETAIL = jest.fn();
const MOCK_SET_POKEMON_DETAIL = jest.fn();

const MOCK_NAVIGATE = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: MOCK_NAVIGATE,
    }),
    useRoute: () => ({
      params: {
        pokemon: 'Pikachu',
      },
    }),
  };
});

const renderWrapper = () =>
  componentRenderer(PokemonDetail)
    .withModels({
      loading: {
        state: {effects: {pokemon: {loadPokemonDetail: false}}},
      },
      pokemon: {
        state: {pokemonDetail: POKEMON_DETAIL},
        effects: () => ({
          loadPokemonDetail: MOCK_LOAD_POKEMON_DETAIL,
          setPokemonDetail: MOCK_SET_POKEMON_DETAIL,
        }),
      },
    })
    .render();

const renderWrapperWithLoadingTrue = () =>
  componentRenderer(PokemonDetail)
    .withModels({
      loading: {
        state: {effects: {pokemon: {loadPokemonDetail: true}}},
      },
      pokemon: {
        state: {pokemonDetail: undefined},
        effects: () => ({
          loadPokemonDetail: MOCK_LOAD_POKEMON_DETAIL,
          setPokemonDetail: MOCK_SET_POKEMON_DETAIL,
        }),
      },
    })
    .render();

let wrapper: any;
describe('PokemonDetail', () => {
  beforeEach(() => {
    wrapper = renderWrapper();
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('when data is loading', () => {
    wrapper = renderWrapperWithLoadingTrue();
    expect(wrapper).toMatchSnapshot();
  });
});
