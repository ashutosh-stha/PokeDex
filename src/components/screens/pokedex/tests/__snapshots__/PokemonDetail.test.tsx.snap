// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetail should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "url",
                    }
                  }
                  style={
                    {
                      "height": 150,
                      "resizeMode": "stretch",
                      "width": 150,
                    }
                  }
                />
                <View
                  style={
                    [
                      {
                        "flexDirection": "column",
                      },
                      {
                        "justifyContent": "center",
                        "margin": 16,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "textTransform": "capitalize",
                      }
                    }
                  >
                    Pikachu
                  </Text>
                  <Text
                    style={
                      {
                        "fontSize": 14,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Type
                  </Text>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "marginRight": 8,
                          "marginTop": 6,
                          "textTransform": "capitalize",
                        }
                      }
                    >
                      electric
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`PokemonDetail when data is loading 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignSelf": "center",
                    "height": 40,
                    "position": "relative",
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                    "width": 40,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "black",
                      "borderRadius": 10,
                      "height": 10,
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "black",
                      "borderRadius": 10,
                      "height": 10,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "black",
                      "borderRadius": 10,
                      "bottom": 0,
                      "height": 10,
                      "left": 0,
                      "position": "absolute",
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "black",
                      "borderRadius": 10,
                      "bottom": 0,
                      "height": 10,
                      "position": "absolute",
                      "right": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
